feed/test.py

```diff
@@ -1,6 +1,65 @@
-"""
-This file demonstrates writing tests using the unittest module. These will pass
-when you run "manage.py test".
+from django.contrib.auth.models import User
+from django.test import TestCase
+from django.test.utils import override_settings
+from rest_framework.test import APIRequestFactory
 
-Replace this with more appropriate tests for your application.
-"""
+from feed.views import ChecklistViewSet
+from workflow.models import Checklist, Organization, TolaUser, WorkflowLevel1, WorkflowLevel2
+
+
+class ViewsTest(TestCase):
+    def setUp(self):
+        self.user = User.objects.create_user('john', 'lennon@thebeatles.com', 'johnpassword')
+        self.user.is_superuser = True
+        self.user.is_staff = True
+        self.user.save()
+
+        wflvl1 = WorkflowLevel1.objects.create(name='WorkflowLevel1')
+        wflvl2 = WorkflowLevel2.objects.create(name='WorkflowLevel2', workflowlevel1=wflvl1)
+        Checklist.objects.bulk_create([
+            Checklist(name='CheckList_0', workflowlevel2=wflvl2),
+            Checklist(name='CheckList_1', workflowlevel2=wflvl2),
+        ])
+
+    def test_list_checklist_superuser(self):
+        factory = APIRequestFactory()
+        request = factory.get('/api/checklist/')
+        request.user = self.user
+        view = ChecklistViewSet.as_view({'get': 'list'})
+        response = view(request)
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(len(response.data), 2)
+
+    def test_list_checklist_normaluser(self):
+        self.user.is_superuser = False
+        self.user.is_staff = False
+        self.user.save()
+        organization = Organization.objects.create(name="TestOrg")
+        TolaUser.objects.create(user=self.user, organization=organization)
```

Comment:
You're creating a normal user in the other test case too. It's better to create it in the setUp, so you don't need to do it again in all the cases.

Reply:
We can't do that because we have `test_list_checklist_superuser`, which is checking that the user is a superuser and does not belong to any org.

Reply:
For this case, I wanted to say to do something like this:

```py
def setUp(self):
    self.super_user = User.objects.create_user('john', 'lennon@thebeatles.com', 'johnpassword')
    self.super_user.is_superuser = True
    self.super_user.is_staff = True
    self.super_user.save()

    self.normal_user = User.objects.create_user('paul', 'mccartney@thebeatles.com', 'paulpassword')
    self.normal_user.is_superuser = False
    self.normal_user.is_staff = False
    self.normal_user.save()
    organization = Organization.objects.create(name="TestOrg")
    TolaUser.objects.create(user=self.user, organization=organization)
```

So, you can use the normal user for the two test cases. As unit tests are executed sequentially, you still have `(len(response.data), 0)` in _test_list_checklist_normaluser_ and `(len(response.data), 1)` in _test_list_checklist_normaluser_one_result_